### ISSUE TITLE
`RANSAC(prosac_sampling=True)` has no effect: the flag is stored in `__init__` and never read

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4869]
`RANSAC.__init__` documents `prosac_sampling: whether to use PROSAC sampling instead of random sampling`, but the flag is only stored. `sample()` always draws uniformly (`torch.rand(batch_size, pop_size).topk(sample_size)`). Its `iteration` argument, documented as "current iteration number (used for PROSAC sampling)", only offsets the seed. These three lines are every occurrence of the name in `kornia/geometry/ransac.py`:

```text
$ grep -n prosac kornia/geometry/ransac.py
65:        prosac_sampling: bool = False,
79:            prosac_sampling: whether to use PROSAC sampling instead of random sampling.
100:        self.prosac_sampling = prosac_sampling
```

PROSAC (Chum & Matas, CVPR 2005) draws its early samples from the top-ranked correspondences and widens the pool progressively. With the true matches ranked first, `prosac_sampling=True` and `False` give identical draws, models and masks.

## Repro

```python
import torch
from kornia.geometry.linalg import transform_points
from kornia.geometry.ransac import RANSAC

H = torch.tensor([[[1.034685, -0.04160598, -63.61482],
                   [0.1039753, 1.135451, -62.87412],
                   [2.851519e-04, 1.404178e-04, 1.0]]])
gx, gy = torch.tensor([160.0, 230.0, 330.0, 440.0]), torch.tensor([95.0, 150.0, 240.0, 320.0])
p1 = torch.stack(torch.meshgrid(gx, gy, indexing="xy"), -1).reshape(16, 2) + torch.linspace(0, 3, 16)[:, None]
p2 = transform_points(H, p1[None])[0]
g = torch.Generator().manual_seed(1)
o1, o2 = torch.rand(16, 2, generator=g) * 400 + 50, torch.rand(16, 2, generator=g) * 400 + 50
# PROSAC assumes the correspondences are sorted by quality: put the 16 true matches first
kp1, kp2 = torch.cat([p1, o1]), torch.cat([p2, o2])

a = RANSAC("homography", seed=3, prosac_sampling=True, max_iter=3, batch_size=16)
b = RANSAC("homography", seed=3, prosac_sampling=False, max_iter=3, batch_size=16)
Ha, ma = a(kp1, kp2)
Hb, mb = b(kp1, kp2)
print("prosac_sampling=True vs False: same model", torch.equal(Ha, Hb), "| same mask", torch.equal(ma, mb))
sa, sb = a.sample(4, 32, 1000, 0), b.sample(4, 32, 1000, 0)
print("sample(): identical draws", torch.equal(sa, sb),
      f"| samples drawn only from the 16 best-ranked matches: {(sa < 16).all(1).float().mean().item():.3f}"
      " (uniform sampling expects C(16,4)/C(32,4) = 0.051)")
```

```text
prosac_sampling=True vs False: same model True | same mask True
sample(): identical draws True | samples drawn only from the 16 best-ranked matches: 0.037 (uniform sampling expects C(16,4)/C(32,4) = 0.051)
```

kornia `main` 7ddf731bd, torch 2.14.0 (and 2.5.1: same output), CPU float32, macOS arm64.

The flag and its docstring were added in #3458; no code path in `kornia/` reads it.

## Classification

API contract gap: a documented constructor option that the code never reads.

No idiom of 1–9; candidate: a documented option that is stored and never read (nearest: 9, serialized state never read back).

## Disposition

Focused fix, welcome as a PR: implement PROSAC's progressive sampling in `sample()` when the flag is set, taking the input order as the quality order (OpenCV's `USAC_PROSAC` makes the same assumption and expects matches sorted by quality). Until then, `__init__` warns that the flag has no effect when it is `True`.

Found by the batch-8 conventions audit (two-view geometry); a `test_wart_*` pin in the docs PR flips when the fix lands.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
